### PR TITLE
Fix Trip edit/delete

### PR DESCRIPTION
**Pull request by rlrh**

### CODE REVIEW

File: TourMate/TourMate/Controllers/FirebaseTripController.swift [hunk 1]
```diff
@@ -32,6 +32,23 @@ struct FirebaseTripController: TripController {
...
+        }
+
+        let (adaptedTrip, errorMessage) = await firebasePersistenceManager.fetchItem(id: tripId)
+        guard let adaptedTrip = adaptedTrip as? FirebaseAdaptedTrip else {
+            return (nil, errorMessage)
+        }
+        guard adaptedTrip.attendeesUserIds.contains(user.uid) else {
+            return (nil, errorMessage)
+        }
+
```
Comment (tanruiquan) [inline TourMate/TourMate/Controllers/FirebaseTripController.swift]: I think we shouldn't reuse the errorMessage if the persistenceManager manages to fetch something right?
```suggestion
        guard let errorMessage.isEmpty else {
           return (nil, errorMessage)
        }
        guard let adaptedTrip = adaptedTrip as? FirebaseAdaptedTrip else {
            return (nil, "[FirebaseTripController] Error converting FirebaseAdaptedData into FirebaseAdaptedTrip")
        }
        guard adaptedTrip.attendeesUserIds.contains(user.uid) else {
            return (nil, "[FirebaseTripController] Error user \(user.uid) is not an attendee of trip \(tripId)")
        }

```

File: TourMate/TourMate/Views/Trip/TripView.swift [hunk 2]
```diff
@@ -8,67 +8,93 @@
 import SwiftUI
 
 struct TripView: View {
+    @Environment(\.dismiss) var dismiss
+
     @StateObject var plansViewModel: PlansViewModel
+    @StateObject var viewModel: TripViewModel
```
Comment (tanruiquan) [inline TourMate/TourMate/Views/Trip/TripView.swift:13]: Maybe can rename to `tripViewModel` cause we have 2 view models here

**Review by tanruiquan — APPROVED** [review]
LGTM 👍

File: TourMate/TourMate/Controllers/FirebaseTripController.swift (hunk 1 above)
Comment (rlrh) [reply TourMate/TourMate/Controllers/FirebaseTripController.swift]: Right!

File: TourMate/TourMate/Views/Trip/TripView.swift (hunk 2 above)
Comment (rlrh) [reply TourMate/TourMate/Views/Trip/TripView.swift:13]: Ok I will try moving PlansViewModel to PlansListView

File: TourMate/TourMate/Views/Trip/TripViewModel.swift [hunk 3]
```diff
@@ -26,7 +26,12 @@ class TripViewModel: ObservableObject {
     func fetchTrip() async {
         self.isLoading = true
         let (trip, errorMessage) = await tripController.fetchTrip(withTripId: trip.id)
-        guard let trip = trip, errorMessage.isEmpty else {
+        guard let trip = trip else {
+            self.isDeleted = true
+            self.hasError = true
+            return
+        }
+        guard errorMessage.isEmpty else {
```
Comment (tanruiquan) [inline TourMate/TourMate/Views/Trip/TripViewModel.swift:34]: I think you should put the guard clause for `.isEmpty` above to catch the `isEmpty` case first

**Review by tanruiquan — APPROVED** [review]
Looks good!

**Pull request merged**